plugins/module_utils/openshift_adm_prune_deployments.py

```diff
@@ -0,0 +1,139 @@
+        self.exit_json = self.module.exit_json
+
+        if not HAS_KUBERNETES_COLLECTION:
+            self.module.fail_json(
+                msg="The kubernetes.core collection must be installed",
+                exception=K8S_COLLECTION_ERROR,
+                error=to_native(k8s_collection_import_exception),
+            )
+
+        super(OpenShiftAdmPruneDeployment, self).__init__(self.module)
+
+        self.params = self.module.params
+        self.check_mode = self.module.check_mode
+        self.client = get_api_client(self.module)
+
+    def filter_replication_controller(self, replicacontrollers):
+        def _deployment(obj):
+            return get_deploymentconfig_for_replicationcontroller(obj) is not None
+
+        def _zeroReplicaSize(obj):
+            return obj['spec']['replicas'] == 0 and obj['status']['replicas'] == 0
+
+        def _complete_failed(obj):
+            DeploymentStatusAnnotation = "openshift.io/deployment.phase"
+            try:
+                # validate that replication controller status is either 'Complete' or 'Failed'
+                deployment_phase = obj['metadata']['annotations'].get(DeploymentStatusAnnotation)
+                return deployment_phase in ('Failed', 'Complete')
+            except Exception:
+                return False
+
+        def _younger(obj):
+            creation_timestamp = datetime.strptime(obj['metadata']['creationTimestamp'], '%Y-%m-%dT%H:%M:%SZ')
+            now = datetime.now(timezone.utc).replace(tzinfo=None)
+            age = (now - creation_timestamp).seconds / 60
+            return age > self.params['keep_younger_than']
+
+        def _orphan(obj):
+            try:
+                # verify if the deploymentconfig associated to the replication controller is still existing
+                deploymentconfig_name = get_deploymentconfig_for_replicationcontroller(obj)
+                exists = self.k8s_ansible_mixin.kubernetes_facts(kind="DeploymentConfig",
+                                                                 api_version="apps.openshift.io/v1",
+                                                                 name=deploymentconfig_name,
+                                                                 namespace=self.params.get('namespace'))
+                return not (exists.get['api_found'] and len(exists['resources']) > 0)
+            except Exception:
+                return False
+
+        predicates = [_deployment, _zeroReplicaSize, _complete_failed]
+        if self.params['orphans']:
+            predicates.append(_orphan)
+        if self.params['keep_younger_than']:
+            predicates.append(_younger)
+
+        results = []
+        for item in replicacontrollers:
+            match = True
+            for pred in predicates:
+                if not pred(item):
+                    match = False
+                    break
+            if match:
+                results.append(item)
+        return results
```

Comment:
Totally optional, but this could probably be simplified by using `filter`:
```python
results = replicacontrollers.copy()
for p in predicates:
    results = filter(p, results)
return list(results)
```